ServerCore/Areas/Identity/UserAuthorizationPolicy.cs

```diff
@@ -8,6 +8,40 @@
 
 namespace ServerCore.Areas.Identity
 {
+    public class IsPlayerInEventRequirement : IAuthorizationRequirement
+    {
+        public IsPlayerInEventRequirement()
+        {
+        }
+    }
+
+    public class IsPlayerInEventHandler : AuthorizationHandler<IsPlayerInEventRequirement>
+    {
+        private readonly PuzzleServerContext puzzleContext;
+        private readonly UserManager<IdentityUser> userManager;
+
+        public IsPlayerInEventHandler(PuzzleServerContext pContext, UserManager<IdentityUser> manager)
+        {
+            puzzleContext = pContext;
+            userManager = manager;
+        }
+
+        protected override Task HandleRequirementAsync(AuthorizationHandlerContext context,
+                                                       IsPlayerInEventRequirement requirement)
+        {
+            PuzzleUser puzzleUser = PuzzleUser.GetPuzzleUserForCurrentUser(puzzleContext, context.User, userManager);
```

Comment:
Is there any way to get here with a logged in liveid, but no PuzzleUser?

Reply:
I don't think so.